api/kv_v2.go

```diff
@@ -399,3 +465,92 @@ func extractFullMetadata(secret *Secret) (*KVMetadata, error) {
 
 	return metadata, nil
 }
+
+func mergePatch(ctx context.Context, client *Client, mountPath string, secretPath string, newData map[string]interface{}, opts ...KVOption) (*KVSecret, error) {
+	pathToMergePatch := fmt.Sprintf("%s/data/%s", mountPath, secretPath)
+
+	// take any other additional options provided
+	// and pass them along to the patch request
+	wrappedData := map[string]interface{}{
+		"data": newData,
+	}
+	options := make(map[string]interface{})
+	for _, opt := range opts {
+		k, v := opt()
+		options[k] = v
+	}
+	if len(opts) > 0 {
+		wrappedData["options"] = options
+	}
```

Comment:
The KVv2 API ignores unexpected options but just wanted to call out that this pattern would be potentially providing things that might not be deemed valid in the future.

Reply:
True, but in that case I would hope that the server would just respond with some kind of invalid option 400 error in that scenario so the user would know that their request had failed and why.

Reply:
Yup, it would in the future. I just didn't know if we wanted to potentially future-proof by filtering it out. I don't think it's strictly necessary _today_ so mostly was pointing it out.